crypto/s2n_ecdsa.c

```diff
@@ -59,7 +83,12 @@ int s2n_ecdsa_sign_digest(const struct s2n_pkey *priv, struct s2n_blob *digest,
     POSIX_ENSURE_REF(key->ec_key);
 
     unsigned int signature_size = signature->size;
-    POSIX_GUARD_OSSL(ECDSA_sign(S2N_ECDSA_TYPE, digest->data, digest->size, signature->data, &signature_size, key->ec_key), S2N_ERR_SIGN);
+
+    /* Safety: ECDSA_sign does not mutate the key */
+    EC_KEY *ec_key = NULL;
+    POSIX_GUARD_RESULT(s2n_unsafe_ecdsa_get_mut(key, &ec_key));
```

Comment:
This might be one of those exceptions to our "always return S2N_RESULT" rules. I liked how your original solution of casting in place meant that we never stored the non-const key, even as a stack variable.

Reply:
spoke offline. this makes sense since its more likely to prevent later use of `*ec_key` past this usage.

Reply:
https://github.com/aws/s2n-tls/pull/3474/commits/0070d164f9dfa7eb7f22251abec6e87f63d663ad